FirebaseVertexAI/Tests/TestApp/Tests/Integration/IntegrationTests.swift

```diff
@@ -235,6 +239,30 @@ final class IntegrationTests: XCTestCase {
       XCTAssertTrue(String(describing: error).contains("Firebase App Check token is invalid"))
     }
   }
+
+  // MARK: - Imagen
+
+  func testGenerateImage_inlineData() async throws {
+    let imagePrompt = """
+    A realistic photo of a male lion, mane thick and dark, standing proudly on a rocky outcrop
+    overlooking a vast African savanna at sunset. Golden hour light, long shadows, sharp focus on
+    the lion, shallow depth of field, detailed fur texture, DSLR, 85mm lens.
+    """
+
+    let imageResponse = try await imagenModel.generateImages(prompt: imagePrompt)
+
+    XCTAssertNil(imageResponse.raiFilteredReason)
+    XCTAssertEqual(imageResponse.images.count, 1)
+    let image = try XCTUnwrap(imageResponse.images.first)
+
+    let textResponse = try await model.generateContent(
+      InlineDataPart(data: image.data, mimeType: "image/png"),
+      "What is the name of this animal? Answer with the animal name only."
+    )
+
+    let text = try XCTUnwrap(textResponse.text).trimmingCharacters(in: .whitespacesAndNewlines)
+    XCTAssertEqual(text, "Lion")
```

Comment:
Can you count on the capitalization?

Reply:
I can't guarantee it though I've found it to be quite deterministic in manual testing with the generation config I'm using for the integration tests:
https://github.com/firebase/firebase-ios-sdk/blob/16381c753c8ff2fe2900fd75b050ac27dacda2ef/FirebaseVertexAI/Tests/TestApp/Tests/Integration/IntegrationTests.swift#L24-L29

Good point to bring up though since some of the other prompts I had experimented with returned "Lion." (i.e., with a period at the end). So it can be quite sensitive.

In a follow-up PR I'll switch this to use `"text/x.enum"` as the `responseMIMEType` and provide an enum `Schema` with animals to pick from. We don't have any integration tests that use that MIME type so it'll provide some additional coverage at the same time. Will also add a GCS-based image generation test.

Reply:
Ended up taking a different approach, only checking the image dimensions in https://github.com/firebase/firebase-ios-sdk/pull/14231. I still think adding a test with `"text/x.enum"` would be good to add for a hardcoded image (but testing one thing at a time, Gemini-only in that case).